### PR TITLE
Bind flags as prefixed env variables

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hector@chainguard.dev>

#### Summary

There is an increase in the amount of flags used in our commands. Therefore we should think about binding these flags as environment variables that would help us when using the commands. 

closes https://github.com/sigstore/cosign/issues/2330

#### Release Note

`Allow setting flag values via environment variables`

### CODE REVIEW

File: cmd/cosign/cli/options/root.go [hunk 1]
```diff
@@ -45,3 +51,27 @@ func (o *RootOptions) AddFlags(cmd *cobra.Command) {
...
+			_ = v.BindEnv(f.Name, flagToEnvVar(f.Name))
+		}
+		if !f.Changed && v.IsSet((f.Name)) {
+			val := v.Get(f.Name)
+			_ = cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
+		}
+	})
+}
+
+func flagToEnvVar(f string) string {
```
Comment (priyawadhwa) [inline cmd/cosign/cli/options/root.go:74]: Could you add a small unit test for these functions?

**Review by priyawadhwa — COMMENTED** [review]
This looks great @hectorj2f !!

File: cmd/cosign/cli/options/root.go (hunk 1 above)
Comment (hectorj2f) [reply cmd/cosign/cli/options/root.go:74]: Done :)

**Review by cpanato — APPROVED** [review]
looks cool

**Pull request merged**